### PR TITLE
Added measurement display label + cached graphics option

### PR DESCRIPTION
**Pull request by bsy-nicholasw**
+ Optional Label added to measurement props. The standard measurements will report it in the widget if it's defined.
  + Added a MeasurementPropertyHelper to allow externally defined measurements to easily create common properties, starting with the name property.

+ Added handling for drawing cached decorations for measurements (so subclasses dont have to manage their own like AreaMeasurement in the future). 
  + Override decorateCached if your measurement needs to participate.
  + A measurement can override both decorate methods, e.g. dynamic per-frame graphics in the standard decorate function and heavy-weight graphics in decorateCached. 
  + Existing methods invalidating measurement decorations will invalidate cached graphics and new public methods have been added to the manager to invalidate all decorations.

### CODE REVIEW

**Comment by calebmshafer** [comment]
@skirby1996 any idea why the builds aren't triggering?

**Comment by aruniverse** [comment]
ah nicholas you also forked the repo. we need to grant you access to the repo if you dont have it already

**Comment by bsy-nicholasw** [comment]
@aruniverse  Yeah I didn't have access, PRs from forks should still be okay? Right?

**Comment by aruniverse** [comment]
> @aruniverse Yeah I didn't have access, PRs from forks should still be okay? Right?

yea thats fine.

@calebmshafer could you add @bsy-nicholasw to the org?

File: packages/measure-tools/src/api/MeasurementCachedGraphicsHandler.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DecorateContext, Decorator, IModelApp, ScreenViewport } from "@bentley/imodeljs-frontend";
+
+/** Handles forwarding a cached graphics decorate call for measurements. Note this is package internal and not meant to be exposed as public API. */
+export class MeasurementCachedGraphicsHandler implements Decorator {
```
Comment (aruniverse) [inline packages/measure-tools/src/api/MeasurementCachedGraphicsHandler.ts:8]: Another static :/

File: packages/measure-tools/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bentley/measure-tools-react",
-  "version": "0.4.1",
+  "version": "0.5.0",
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply packages/measure-tools/package.json:3]:
```suggestion
  "version": "0.4.1",
```

**Comment by aruniverse** [comment]
LGTM, only change id like to request is revert the manual version bump

**Pull request merged**